### PR TITLE
feat(tests): Split pubsub integration tests for faster execution

### PR DESCRIPTION
**Pull request by ttomsu**
Splits the `pubsub` module up for faster execution. Takes it from ~15min to ~11-12min.

### CODE REVIEW

File: spring-cloud-gcp-samples/spring-cloud-gcp-integration-pubsub-sample/spring-cloud-gcp-integration-pubsub-sample-receiver/src/test/java/com/example/ReceiverIntegrationTest.java [hunk 1]
```diff
@@ -61,7 +61,7 @@
 	@BeforeClass
 	public static void prepare() {
 		assumeThat(
-				"PUB/SUB-sample integration tests are disabled. Please use '-Dit.pubsub=true' "
+				"PUB/SUB-sample integration tests are disabled. Please use '-Dit.pubsub-integration=true' "
 						+ "to enable them. ",
 				System.getProperty("it.pubsub"), is("true"));
```
Comment (elefeint) [inline spring-cloud-gcp-samples/spring-cloud-gcp-integration-pubsub-sample/spring-cloud-gcp-integration-pubsub-sample-receiver/src/test/java/com/example/ReceiverIntegrationTest.java:66]: property needs to be updated to match message

File: spring-cloud-gcp-samples/spring-cloud-gcp-integration-pubsub-sample/spring-cloud-gcp-integration-pubsub-sample-polling-receiver/src/test/java/com/example/PollingReceiverIntegrationTest.java [hunk 2]
```diff
@@ -60,7 +60,7 @@
 	@BeforeClass
 	public static void prepare() {
 		assumeThat(
-				"PUB/SUB-sample integration tests are disabled. Please use '-Dit.pubsub=true' "
+				"PUB/SUB-sample integration tests are disabled. Please use '-Dit.pubsub-integration=true' "
 						+ "to enable them. ",
 				System.getProperty("it.pubsub"), is("true"));
```
Comment (elefeint) [inline spring-cloud-gcp-samples/spring-cloud-gcp-integration-pubsub-sample/spring-cloud-gcp-integration-pubsub-sample-polling-receiver/src/test/java/com/example/PollingReceiverIntegrationTest.java:65]: property needs to be updated to match message

File: spring-cloud-gcp-samples/spring-cloud-gcp-integration-pubsub-sample/spring-cloud-gcp-integration-pubsub-sample-sender/src/test/java/com/example/SenderIntegrationTest.java [hunk 3]
```diff
@@ -60,7 +60,7 @@
 	@BeforeClass
 	public static void prepare() {
 		assumeThat(
-				"PUB/SUB-sample integration tests are disabled. Please use '-Dit.pubsub=true' "
+				"PUB/SUB-sample integration tests are disabled. Please use '-Dit.pubsub-integration=true' "
```
Comment (elefeint) [inline spring-cloud-gcp-samples/spring-cloud-gcp-integration-pubsub-sample/spring-cloud-gcp-integration-pubsub-sample-sender/src/test/java/com/example/SenderIntegrationTest.java:63]: property needs to be updated to match message

File: spring-cloud-gcp-samples/spring-cloud-gcp-integration-pubsub-sample/spring-cloud-gcp-integration-pubsub-sample-test/src/test/java/com/example/SampleAppIntegrationTest.java [hunk 4]
```diff
@@ -52,7 +52,7 @@
 	@BeforeClass
 	public static void prepare() {
 		assumeThat(
-				"PUB/SUB-sample integration tests are disabled. Please use '-Dit.pubsub=true' "
+				"PUB/SUB-sample integration tests are disabled. Please use '-Dit.pubsub-integration=true' "
```
Comment (elefeint) [inline spring-cloud-gcp-samples/spring-cloud-gcp-integration-pubsub-sample/spring-cloud-gcp-integration-pubsub-sample-test/src/test/java/com/example/SampleAppIntegrationTest.java:55]: property needs to be updated to match message

File: spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-bus-config-sample/spring-cloud-gcp-pubsub-bus-config-sample-test/src/test/java/com/example/LocalSampleAppIntegrationTest.java [hunk 5]
```diff
@@ -76,7 +76,7 @@
 	@BeforeClass
 	public static void prepare() throws Exception {
 		assumeThat(
-			"PUB/SUB integration tests are disabled. Use '-Dit.pubsub=true' to enable.",
+			"PUB/SUB integration tests are disabled. Use '-Dit.pubsub-bus=true' to enable.",
```
Comment (elefeint) [inline spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-bus-config-sample/spring-cloud-gcp-pubsub-bus-config-sample-test/src/test/java/com/example/LocalSampleAppIntegrationTest.java:79]: property needs to be updated to match message

**Review by elefeint — CHANGES_REQUESTED** [review]
It might get even faster once the integration property flag is updated. Or did you have an unpushed commit?

Overall, this makes it a tiny bit harder to test all pubsub manually after making a change, but I think that's okay. I always wished for master switch like `it.all`; maybe we can eventually have one like `it.all` and `it.pubsub` for manual testing.

**Comment by ttomsu** [comment]
D'oh! Great catch - rerunning again for real numbers.

**Comment by ttomsu** [comment]
10m20s - I can still do better!

**Review by dzou — APPROVED** [review]

File: .github/workflows/buildAndTest.yaml [hunk 6]
```diff
@@ -94,7 +94,10 @@ jobs:
           - metrics
           - kotlin
           - pubsub
```
Comment (meltsufin) [inline .github/workflows/buildAndTest.yaml:96]: Do we still have the regular `it.pubsub=true` tests?

**Review by meltsufin — APPROVED** [review]

**Review by elefeint — APPROVED** [review]

File: .github/workflows/buildAndTest.yaml (hunk 6 above)
Comment (ttomsu) [reply .github/workflows/buildAndTest.yaml:96]: Yes - for fairly basic sample integration tests

**Pull request merged**